### PR TITLE
fix: fix and optimize tls in upstream_schema

### LINKED ISSUE #10260: bug: wrong json schema configuration
### Current Behavior

apisix/schema_def.lua:414
                client_cert_id = {
                    ["not"] = {required = {"client_client", "client_key"}}
                }
"client_client" has never appeared

### Expected Behavior

_No response_

### Error Logs

_No response_

### Steps to Reproduce

apisix/schema_def.lua:414
                client_cert_id = {
                    ["not"] = {required = {"client_client", "client_key"}}
                }
maybe "client_client" here should be "client_cert"

### Environment

- APISIX version (run `apisix version`):2.15(The latest version also has the same issue)
- Operating system (run `uname -a`):Linux 4.14.105-1-tlinux3-0022 #1 SMP Wed Sep 29 11:01:45 CST 2021 x86_64 x86_64 x86_64 GNU/Linux
- OpenResty / Nginx version (run `openresty -V` or `nginx -V`):openresty/1.21.4.1
- etcd version, if relevant (run `curl http://127.0.0.1:9090/v1/server_info`):
- APISIX Dashboard version, if relevant:
- Plugin runner version, for issues related to plugin runners:
- LuaRocks version, for installation issues (run `luarocks --version`):3.8.0

### PR DESCRIPTION
**Pull request by wzy0618**
### Description
Removed two redundant configurations.
Fixed the wrong variables mentioned in the issue.

Fixes #10260

### Checklist
- [ ] I have explained the need for this PR and the problem it solves
- [ ] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

File: apisix/schema_def.lua [hunk 1]
```diff
@@ -402,16 +402,10 @@ local upstream_schema = {
                 },
             },
             dependencies = {
-                client_cert = {
-                    required = {"client_key"},
-                    ["not"] = {required = {"client_cert_id"}}
```
Comment (monkeyDluffy6017) [inline apisix/schema_def.lua:407]: Why do you remove the not condition?

Comment (monkeyDluffy6017) [reply apisix/schema_def.lua:407]: @soulbird please help to check this

Comment (Revolyssup) [reply apisix/schema_def.lua:407]: The not condition is not required at both places `client_cert` and `client_cert_id`, having the not in clien_cert_id is enough to make both fields mutually exclusive. @monkeyDluffy6017

**Review by Revolyssup — APPROVED** [review]

File: apisix/schema_def.lua (hunk 1 above)
Comment (wzy0618) [reply apisix/schema_def.lua:407]: As @Revolyssup said, there should only be the following three cases of the three properties(client_cert_id, client_cert, client_key) here:
1. None of them exist
2. Only client_cert, client_key
3. Only client_cert_id
The modified conditions can ensure that "client_cert" and "client_key" must exist at the same time, and three properties will not exist at the same time.

Comment (monkeyDluffy6017) [reply apisix/schema_def.lua:407]: Could you add some test cases to cover these conditions?

**Comment by monkeyDluffy6017** [comment]
hi @wzy0618 , are you still working on this?

**Comment by wzy0618** [comment]
> hi @wzy0618 , are you still working on this?

yeah, I just finished my holiday and I will submit it this week.

**Review by monkeyDluffy6017 — APPROVED** [review]

**Review by AlinsRan — APPROVED** [review]

**Pull request merged**